snapcraft/internal/errors.py

```diff
@@ -42,15 +39,61 @@ def get_exit_code(self):
 
 
 class SnapcraftReportableError(SnapcraftError):
-    """Base class for all snapcraft exceptions that integrate with Sentry.
+    """DEPRECATED: Use SnapcraftAbstractError instead with reportable=True."""
 
-    An exception class inheriting from this will get a prompt to report
-    to sentry if raised and is handled by the exception handler on program
-    exit.
 
-    :cvar fmt: A format string that daughter classes override
+class SnapcraftAbstractError(SnapcraftError, ABC):
+    """Abstract class for all snapcraft exceptions."""
 
-    """
+    @property
+    @abstractmethod
+    def brief(self) -> str:
+        """Concise, single-line description of the error."""
+        pass
+
+    @property
+    def details(self) -> str:
+        """Detailed information, if considered necessary for user."""
+        return None
```

Comment:
```suggestion
        return ""
```

Or change the return annotation to `Optional[str]`